src/main/java/htsjdk/samtools/util/AbstractProgressLogger.java

```diff
@@ -80,10 +89,10 @@ public synchronized boolean log() {
         }
     }
 
-    @Override
-    public synchronized boolean record(final String chrom, final int pos) {
+    protected synchronized boolean record(final String chrom, final int pos, final String rname) {
```

Comment:
Why not making this public API also in the `ProgressLoggerInterface`? Instead of read name, it can also log record names (might be useful for some other data sources). For the interface, you can use a default value not to break compatibility with other implementations...

Reply:
this is clearly scope-creep. I would hold-off on requesting that from the OP